lib/sequel/adapters/vertica.rb

```diff
@@ -126,9 +126,33 @@ def schema_parse_table(table_name, options = {})
 
     class Dataset < Sequel::Dataset
       Database::DatasetClass = self
-      EXPLAIN = 'EXPLAIN '
-      EXPLAIN_LOCAL = 'EXPLAIN LOCAL '
-      QUERY_PLAN = 'QUERY PLAN'
+      EXPLAIN = 'EXPLAIN '.freeze
+      EXPLAIN_LOCAL = 'EXPLAIN LOCAL '.freeze
+      QUERY_PLAN = 'QUERY PLAN'.freeze
+      TIMESERIES = ' TIMESERIES '.freeze
+      OVER = ' OVER '.freeze
+      AS = ' AS '.freeze
+
+      Dataset.def_sql_method(self, :select, %w(with select distinct columns from join timeseries where group having compounds order limit lock))
+
+      def timeseries(opts={})
+        fail "timeseries requires :alias" unless opts[:alias]
+        fail "timeseries requires :time_unit" unless opts[:time_unit]
+        fail "timeseries requires an :over clause" unless opts[:over]
```

Comment:
Gah, sorry, this response is so late. I'll fix this. At the time, rubocop had taken the position that `fail` was preferred to `raise` and I was supporting that. They've since reversed their position, and I see no reason not to use `ArgumentError` here.